elements/lisk-p2p/src/p2p_types.ts

```diff
@@ -77,6 +78,7 @@ export interface P2PNodeInfo extends P2PSharedState {
 	readonly os: string;
 	readonly nethash: string;
 	readonly wsPort: number;
+	readonly advertiseAddress?: boolean;
```

Comment:
Make this non-optional.

Reply:
as discussed with Ishan, by default the advertiseAddress is handled as true. I understand your intention behind avoid the extra check, but as we discussed, for p2p it's optional and its a configuration from the framework, keeping it optional would be fine in my opinion. Same for below.

Reply:
The final discussion was that it's ok to make it optional when passing the conf value but it's better to make it mandatory internally. Please, make it non-optional in `P2PNodeInfo` iface.

Reply:
In general, we should avoid optional fields overall when they are booleans. It doesn't make sense to have an optional boolean...